components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -68,14 +69,19 @@ class SitePermissionsFeature(
 ) : LifecycleAwareFeature {
 
     private val observer = SitePermissionsRequestObserver(sessionManager, feature = this)
-    internal val ioCoroutineScope by lazy { CoroutineScope(Dispatchers.IO) }
+    internal val ioCoroutineScope by lazy { coroutineScopeInitializer() }
+
+    internal var coroutineScopeInitializer = {
+        CoroutineScope(Dispatchers.IO)
+    }
 
     override fun start() {
         observer.observeSelected()
     }
 
     override fun stop() {
         observer.stop()
+        ioCoroutineScope.cancel()
```

Comment:
@pocmo I also added a this to cancel all the coroutines when the feature is stopped

Reply:
Doesn't this mean the feature won't work anymore on the next start? Or can you continue to use the scope after cancelling?

Reply:
Good point, I will do some research!

Reply:
I will address that in https://github.com/mozilla-mobile/android-components/issues/2889. I will remove this part from this PR

Reply:
👍